### PR TITLE
Suggestion - !url

### PR DESCRIPTION
**Pull request by LW001**
Added !url UVID which parses an ID into a link. Also validates it to avoid typos.

### CODE REVIEW

**Comment by Arzte** [comment]
Why is this needed, why does it need to interact with uservoice can't you just pass the Id to the end of a URL?

**Review by Dougley — CHANGES_REQUESTED** [review]
~~`Fix Travis` commit but Travis still fails lmao~~
Please fix your style issues first.

File: Commands/Commands/uservoice.js [hunk 1]
```diff
@@ -245,11 +245,9 @@ commands.url = {
...
-    uvClient.v1.loginAsOwner().then(c => {
       let id = suffix.match(/\n{9}/)
-      c.get(`forums/${config.uservoice.forumId}/suggestions/${id}.json`).then((s) => {
-     msg.reply(`I was able to find that suggestion! Here's the link: ${s.suggestion.url}`)
-   }).catch(e => {
+      if (id[1]!==null) msg.reply(`Here's the link: https://${config.uservoice.subdomain}.${config.uservoice.domain}/forums/${config.uservoice.forumId}/suggestions/${id}`)
+      else msg.reply(`Please enter a Uservoice ID.`)
      if (e.statusCode === 404) {
        msg.reply(`Sadly I couldn't find that suggestion, though if you still need the link, here it is: https://${config.uservoice.subdomain}.${config.uservoice.domain}/forums/${config.uservoice.forumId}/suggestions/${id}`)
      } else {
```
Comment (Arzte) [inline Commands/Commands/uservoice.js]: ahem, you missed some useless stuff

File: Commands/Commands/uservoice.js [hunk 2]
```diff
@@ -243,32 +243,10 @@ commands.comment = {
...
-     logger.log(bot, {
-       cause: 'login_as',
-       message: (e.message !== undefined) ? e.message : JSON.stringify(e)
-     }, e)
-     msg.reply('an error occured, please try again later.').then(errmsg => {
-       setTimeout(() => bot.Messages.deleteMessages([msg, errmsg]), config.timeouts.errorMessageDelete)
-     })
-   })
+    if (suffix.length === 9) msg.reply(`Here's the link: https://${config.uservoice.subdomain}.${config.uservoice.domain}/forums/${config.uservoice.forumId}/suggestions/${suffix}`)
+    else msg.reply(`Please enter a Suggestion ID.`)
```
Comment (nullpixel) [inline Commands/Commands/uservoice.js]: you should use the UVRegex:tm:

**Review by nullpixel — CHANGES_REQUESTED** [review]

File: Commands/Commands/uservoice.js (hunk 2 above)
Comment (LW001) [reply Commands/Commands/uservoice.js]: What's the use in taking a URL for a command that returns a URL though

Comment (nullpixel) [reply Commands/Commands/uservoice.js]: I mean to validate the ID

File: Commands/Commands/uservoice.js [hunk 3]
```diff
@@ -240,6 +240,15 @@ commands.comment = {
     })
   }
 }
+commands.url = {
+  adminOnly: false,
+  modOnly: false,
+  fn: function (bot, msg, suffix) {
+    msg.channel.sendTyping()
```
Comment (Dougley) [inline Commands/Commands/uservoice.js]: Sending typing is trivial because race conditions, I'd say remove it.

**Review by Dougley — DISMISSED** [review]

File: Commands/Commands/uservoice.js (hunk 2 above)
Comment (LW001) [reply Commands/Commands/uservoice.js]: We dropped that because UV is too slow ty

**Review by nullpixel — APPROVED** [review]

**Comment by nullpixel** [comment]
I still don't see the point of this though

**Pull request merged**